sklearn/utils/estimator_checks.py

```diff
@@ -878,15 +878,27 @@ def check_array_api_input(
     X_xp = xp.asarray(X, device=device)
     y_xp = xp.asarray(y, device=device)
 
-    est.fit(X, y)
+    X_types = est._get_tags().get("X_types", [""])
+    if "labels" in X_types[0]:
+        fit_args = (y,)
+        xp_fit_args = (y_xp,)
+        method_arg = (y,)
+        xp_method_arg = (y_xp,)
+    else:
+        fit_args = (X, y)
+        xp_fit_args = (X_xp, y_xp)
+        method_arg = (X,)
+        xp_method_arg = (X_xp,)
```

Comment:
@jeromedockes any idea why you did not need to introduce something similar in #28626?